backend/stylists/app/Http/Controllers/Merchant/ProductController.php

```diff
@@ -27,24 +30,34 @@ public function index(Request $request, $action)
         return $this->$method($request);
     }
 
-    public function initWhereConditions(Request $request){
+    public function initWhereConditions(Request $request)
+    {
         $this->where_conditions['is_moderated'] = false;
         parent::initWhereConditions($request);
     }
 
-    public function postList(Request $request){
+    public function postList(Request $request)
+    {
         $this->initWhereConditions($request);
-        if($request->input('approve_all')){
-            $this->approveAllProducts();
-        }
-        elseif($request->input('reject_all')){
-            $this->rejectAllProducts();
+
+        $selected_products = !empty($request->input('product_ids')) ? explode(',', $request->input('product_ids')) : '';
+        $errorObj = new Error();
+
+        if (!empty($selected_products)) {
+            if ($request->input('approve_all')) {
+                $response = $this->approveAllProducts($selected_products);
+            } elseif ($request->input('reject_all')) {
+                $response = $this->rejectAllProducts($selected_products);
+            }
+        } else {
+            $response = $errorObj->error(405);
         }
 
```

Comment:
Add the above code in try catch
In try add DB::commit();
in Catch add DB::rollBack();